### DISCUSSION TITLE
MQT: Connect failed to core-mosquitto:1883, rc -5. Retry in 10 sec

### DISCUSSION THREAD

**Discussion by Sherlock2004** [arendst/Tasmota#21443] [Support request]
Hello.
I'm trying to use a Chinese smart IR remote control under Tasmota. The firmware was installed successfully the first time. But I can't connect to my Home Asistaint. I keep getting an error

```
18:50:15.032 MQT: Connect failed to core-mosquitto:1883, rc -5. Retry in 10 sec
18:50:26.060 MQT: Attempting connection...
18:50:26.074 MQT: Connect failed to core-mosquitto:1883, rc -5. Retry in 20 sec
18:50:47.024 MQT: Attempting connection...
18:50:47.032 MQT: Connect failed to core-mosquitto:1883, rc -5. Retry in 30 sec
18:51:18.015 MQT: Attempting connection...
18:51:18.025 MQT: Connect failed to core-mosquitto:1883, rc -5. Retry in 40 sec
18:51:59.050 MQT: Attempting connection...
18:51:59.062 MQT: Connect failed to core-mosquitto:1883, rc -5. Retry in 50 sec
18:52:50.023 MQT: Attempting connection...
18:52:50.033 MQT: Connect failed to core-mosquitto:1883, rc -5. Retry in 60 sec
18:53:51.043 MQT: Attempting connection...
18:53:51.051 MQT: Connect failed to core-mosquitto:1883, rc -5. Retry in 70 sec
18:55:02.023 MQT: Attempting connection...
18:55:02.032 MQT: Connect failed to core-mosquitto:1883, rc -5. Retry in 80 sec
```

My home assistant is on a mini pc on my local network. It is installed there on the Virtual box virtual machine.

The Mosquitto broker addon is installed there and configured by default (there are few settings at all). MQTT and Tasmota integration is also installed. Judging by the log, MQTT integration connects to Mosquitto broker, but this is not surprising, because they are located on the same computer.

I'm trying to connect there via my home Wi-Fi network. I have an ASUS router installed at home, the only thing done there is that some ports are redirected to this mini-PC.

I am not strong in knowledge of Internet networks. I read that rc -5 is a DNS error or an incorrect name, the name of what, what DNS, here I honestly have no idea what I need to do. Two days of searching on Google yielded no results. 🙄

**Comment by barbudor** [comment]
- Does your Tasmota gets it's IP address from DHCP or did you set a manual IP by yourself ?
 If using manual, you must insure that you have set the IP address of your DNS in `IPAddress4`. Generally the IP address of your router (something like 192.168.?.1 or 192.168.?.254).
 
- How have your configure the MQTT broker in tasmota ? By an IP address or a name ? Which name ?
 If a name, is that name reachable from another PC (not the mini-PC). You can test by `ping thename`. Even if ping return errors, it should give you  the IP address that match the name

- Can you connect to the MQTT broker by a PC tool like MQTT Explorer or MQTTX ?

Note that Tasmota Discussions is NOT a general support forum for computing an networks. You must understand a minimum about networks and IP to use Tasmota in your setup which might be different than from other users.

**Comment by Sherlock2004** [comment]
> * Does your Tasmota gets it's IP address from DHCP or did you set a manual IP by yourself ?
>   If using manual, you must insure that you have set the IP address of your DNS in `IPAddress4`. Generally the IP address of your router (something like 192.168.?.1 or 192.168.?.254).

The device receives an IP automatically, I didn’t even find any settings in the tasmota interface to set a specific IP. 👀

> * How have your configure the MQTT broker in tasmota ? By an IP address or a name ? Which name ?
>   If a name, is that name reachable from another PC (not the mini-PC). You can test by `ping thename`. Even if ping return errors, it should give you  the IP address that match the name

![scr_19_05_2024_21_56_26](https://github.com/arendst/Tasmota/assets/88171363/95f6c908-14fd-469d-ae50-991c7c732dfd)
If I enter the device’s IP from my main PC, I get a response, there are no lost packets, well, in theory, I open the tasmota web interface and see the device logs.

> * Can you connect to the MQTT broker by a PC tool like MQTT Explorer or MQTTX ?

![scr_19_05_2024_21_53_20](https://github.com/arendst/Tasmota/assets/88171363/36e35abb-2562-4294-9aa9-cfaeb1f2c77f)

**Reply by sfromis** [reply]
Have you made sure to setup DNS to have the name `core-mosquitto` resolve to the IP address? If not, do that, or use the IP address instead of an unresolved name.

**Comment by barbudor** [comment]
> If I enter the device’s IP from my main PC, I get a response, there are no lost packets

That is not what I asked
I asked to ping the MQTT broker, in your case 'core-mosquitto'
This is also what sfromis is asking

Beside, why do you set 'core-mosquitto' in Tasmota and 'mosquitto' in mqtt-explorer ?

**Reply by sfromis** [reply]
Well, that part is due to MQTT Explorer having the "Name" just meaning a label for human consumption, while the "Host" name is below that. But yeah, the key is that you need to have your DNS resolve names if you do not want to use "raw" IP addresses.

**Reply by barbudor** [reply]
Ah, yes

**Comment by Sherlock2004** [comment]
> Beside, why do you set 'core-mosquitto' in Tasmota and 'mosquitto' in mqtt-explorer ?

in the first line there is just the name of the entry, and then the address, isn’t it???

**Reply by sfromis** [reply]
Right, as I just posted above :smile:

**Comment by Sherlock2004** [comment]
I'm completely stupid, in HA I don't see anywhere any hint of changing the "domain name - core-mosquitto" to something else, for example IP, secondly, I have no idea how I can redirect "core- mosquitto" to a specific IP using an asus router.

I tried to enter not “core-mosquitto” into the device with tasmota, but specifically the IP address of the mini-PC, after which I began to receive an rc -2 error.
![scr_19_05_2024_22_44_07](https://github.com/arendst/Tasmota/assets/88171363/68e090e5-6b5b-4dde-83a8-a4db406d7b54)

_Sorry. But two days of torment and attempts to find something on forums and discussions gave me nothing. I have several magic home controllers and this IR remote control (I decided to conduct an experiment on it), which I have already reflashed and cannot return to its original state._

**Reply by barbudor** [reply]
from rc-5 to rc-2 , that's progress

Now it shows that tasmota cannot connect to the MQTT broker
What about MQTT explorer (using IP address too) ?

If still not, you probably have a problem with your virtualbox setttings

**Reply by sfromis** [reply]
While I do not specifically know about the UI of Asus routers, the general approach would be to reserve an IP address, with a name. That is likely to be near something like DHCP configuration.

Note that if your Mosquitto setup is new, and using a current Mosquitto version, you need to have Mosquitto configured to even accept connections from the network. As a starting point, you could have this in `mosquitto.conf`:
```
listener 1883 0.0.0.0
allow_anonymous true
```

**Reply by Sherlock2004** [reply]
![scr_20_05_2024_00_02_00](https://github.com/arendst/Tasmota/assets/88171363/0b03864e-ee47-4d7b-add1-854b27ca6293)
All I see in terms of addon settings for HA. Maybe you will see something suitable?
https://github.com/home-assistant/addons/blob/master/mosquitto/DOCS.md

**Reply by sfromis** [reply]
You did not answer the question from barbudor, about whether you can connect MQTT Explorer to Mosquitto. My mention of the possibility of needing config info was predicated on that.

And it is also key to look into the other thing mentioned by barbudor, about possible "incompleteness" of virtualbox settings.

**Reply by Sherlock2004** [reply]
I have already answered, below is a message with a screenshot where I connected via IP, I already wrote that I have no idea how to set up redirection from “text” to some IP in the router, there you can only do this from IP to IP.

**Comment by Sherlock2004** [comment]
I also have HFS server installed on my mini-PC, it opens without problems through the port that I redirect through the router (from the Internet, not locally). I also added all the other ports there, i.e. 1883,1884,8883,8884 which are mentioned in the broker, but this did not bring results.
![scr_19_05_2024_22_58_51](https://github.com/arendst/Tasmota/assets/88171363/ac7950d3-ca94-4fa8-9d91-103bb8833023)

**Comment by Sherlock2004** [comment]
So, I decided to look for a video. I came across a video where mosquitto broker is launched on the Windows machine itself as a service, I’m quite happy with it. I have already checked and devices connect to it without problems and MQTT Explorer too. Apparently there really are some problems with access to the virtual machine.
![scr_20_05_2024_06_09_18](https://github.com/arendst/Tasmota/assets/88171363/46d13727-1d7e-4ec4-94a8-0edf458e5d02)

**Comment by Sherlock2004** [comment]
I understood why it didn't work. I have the network mode "network bridge" configured in my virtual machine. In this mode, the virtual machine becomes a full-fledged computer and requests a separate IP for itself. That is, I didn’t need to connect to the IP mini-PC on which the machine was running, but just look at the IP that the home assistant shows. I checked this IP and immediately connected to the broker that was launched as an addition to the home assistant.
**Perhaps this will be useful to someone...**
![scr_21_05_2024_06_30_23](https://github.com/arendst/Tasmota/assets/88171363/9cbaaa3e-6aab-40a5-806f-7b8f1b0ac7cf)

**Comment by RajaAbdullahSaleh** [comment]
Hello, I want to create my own broker to connect with esp32 can anyone guide which broker to use and how to create it

**Reply by sfromis** [reply]
The Mosquitto project has a good open source broker, simple to install and use.
